src/hotspot/cpu/x86/interp_masm_x86.hpp

```diff
@@ -139,9 +139,19 @@ class InterpreterMacroAssembler: public MacroAssembler {
   // Expression stack
   void pop_ptr(Register r = rax);
   void pop_i(Register r = rax);
+
+  // On x86, pushing a ptr or an int is semantically identical, but we
+  // maintain a distinction for clarity and for making it easier to change
+  // semantics in the future
   void push_ptr(Register r = rax);
   void push_i(Register r = rax);
 
+  // push_i_or_ptr is provided for when explicitly allowing either a ptr or
+  // an int might have some advantage, while still documenting the fact that a
+  // ptr might be pushed to the stack. This method will never do any
+  // verification of the oop.
```

Comment:
I don't think we need to mention verification here.

Reply:
Sure, I'll drop that.